pkg/health/pods.go

```diff
@@ -116,8 +116,8 @@ func PodsUnhealthy(client k8sclient.Client, options *k8sclient.ListOptions) (*[]
 	unhealthy := []unstructured.Unstructured{}
 	for _, pod := range podList.Items {
 		if pod.Status.Phase != corev1.PodFailed &&
-			!containerUnhealthy(pod) &&
-			!unhealthyStatusConditions(pod) {
+			!ContainerUnhealthy(pod) &&
+			!UnhealthyStatusConditions(pod) {
```

Comment:
while re-using code is good, I expected you to create a new function instead of modifying this one. Also, is the `UnhealthyStatusConditions` required?

In the new implementation, I think the following logic will work too:

1. for a pod in all pods, 
2.    for a container in pod.Container
3.         if !container.Ready, return this pods as unhealthy pods


You can get away with not having the extra logic that `ContainerUnhealthy` has

Reply:
Done, updated the PR